include/mbedtls/mbedtls_config.h

```diff
@@ -2800,6 +2800,20 @@
  */
 #define MBEDTLS_GCM_C
 
+/**
+ * \def MBEDTLS_GCM_LARGETABLE
```

Comment:
I am open to merge the option with another one, however, I dont find `MBEDTLS_AES_FEWER_TABLES` the correct choice, as the GCM optimization is not limited to AES block ciphers but all block ciphers used with GCM.

Reply:
Agreed, but in the team we did previously talk about using it slightly more generally as a (poorly named) size/perf trade-off (see some discussion at https://github.com/Mbed-TLS/mbedtls/pull/8530#discussion_r1419975133 )

Reply:
As using large tables has a very high effect on runtime memory requirements, I would keep it separate from other configuration options for now. Maybe, someone wants generally faster AES but does not accept the overhead for GCM. Additionally `MBEDTLS_AES_FEWER_TABLES` is off by default and I intend to keep `MBEDTLS_GCM_LARGETABLE` off as well by default. By merging both parameters, GCM would by default use large tables and only by activating fewer tables, the memory consumption would be reduced to the current state.

Reply:
After discussing with @daverodgman, while having another config option isn't great, the current `MBEDTLS_AES_FEWER_TABLES` can't really be used for this. Which would mean replacing it and `MBEDTLS_GCM_LARGETABLE` with something that covered both, and it would have to have at least three different values. And overall that wouldn't be much less complex that having `MBEDTLS_AES_FEWER_TABLES` and `MBEDTLS_GCM_LARGETABLE` separately.

One thing, though: `MBEDTLS_AES_FEWER_TABLES` has `_` between the adjective `FEWER` and noun `TABLES`, whereas `MBEDTLS_GCM_LARGETABLE` doesn't. Would it be more consistent to have `MBEDTLS_GCM_LARGE_TABLE`?

Reply:
I now changed `MBEDTLS_GCM_LARGETABLE` to `MBEDTLS_GCM_LARGE_TABLE`.